backend/app/src/main/java/ch/zhaw/studyflow/controllers/CalendarController.java

```diff
@@ -0,0 +1,328 @@
+                    .flatMap(calendar -> {
+                        Optional<Long> userId = principal.getClaim(CommonClaims.USER_ID).map(Long::valueOf);
+                        if (userId.isPresent()) {
+                            Calendar foundCalendar = calendarManager.read(calendar.getId(), userId.get());
+                            return Optional.of(foundCalendar);
+                        } else {
+                            return Optional.empty();
+                        }
+                    })
+                    .ifPresentOrElse(
+                            calendar -> {
+                                response.setResponseBody(JsonContent.writableOf(calendar));
+                                response.setStatusCode(HttpStatusCode.OK);
+                            },
+                            () -> {
+                                response.setStatusCode(HttpStatusCode.BAD_REQUEST);
+                            }
+                    );
+            return response;
+        });
+    }
+
+    /**
+     * Endpoint for retrieving all calendars for the authenticated user.
+     *
+     * @param context the request context
+     * @return the HTTP response
+     */
+    @Endpoint(method = HttpMethod.GET)
+    public HttpResponse getCalendars(RequestContext context) {
+        final HttpRequest request = context.getRequest();
+        HttpResponse response = context.getRequest().createResponse();
+        return authenticator.handleIfAuthenticated(request, principal -> {
+            Optional<Long> userId = principal.getClaim(CommonClaims.USER_ID).map(Long::valueOf);
+            if (userId.isPresent()) {
+                List<Calendar> calendars = calendarManager.getCalendarsByUserId(userId.get());
+                response.setResponseBody(JsonContent.writableOf(calendars))
+                        .setStatusCode(HttpStatusCode.OK);
+            } else {
+                response.setStatusCode(HttpStatusCode.BAD_REQUEST);
+            }
+            return response;
+        });
+    }
+
+    /**
+     * Endpoint for retrieving all appointments in a specific calendar within a date range.
+     *
+     * @param context the request context
+     * @return the HTTP response
+     */
+    @Route(path = "{calendarId}/appointments")
+    @Endpoint(method = HttpMethod.GET)
+    public HttpResponse getAppointments(RequestContext context) {
+        final HttpRequest request = context.getRequest();
+        HttpResponse response = context.getRequest().createResponse();
+        return authenticator.handleIfAuthenticated(request, principal -> {
+            Optional<Long> userId = principal.getClaim(CommonClaims.USER_ID).map(Long::valueOf);
+            if (userId.isPresent()) {
+                QueryParameters queryParameters = request.getQueryParameters();
+                Date from = queryParameters.getSingleValue("from")
+                        .map(Date::new)
+                        .orElse(new Date(0));
```

Comment:
I think it would make sense to use the current date - 1 month instead of the 1.1.1970.